src/modules/http.c

```diff
@@ -464,9 +464,16 @@ http_run(void *data)
         agent_daemon_addr.sin_family = AF_INET;
         agent_daemon_addr.sin_port = htons (port);
 
-        inet_pton (AF_INET, addr, &agent_daemon_addr.sin_addr);
-
  	GET_PRIV(core, http);
+
+       int addr_ok = inet_pton (AF_INET, addr, &agent_daemon_addr.sin_addr);
+
+	if (addr_ok <= 0) {
```

Comment:
`addr_ok` can't be `-1` according to the manual, so you could do something like this instead:

```c
addr_ok = inet_pton(AF_INET, ...);

if (!addr_ok)
    addr_ok = inet_pton(AF_INET6, ...);

assert(addr_ok >= 0);
if (!addr_ok)
     /* fail */
```